### ISSUE TITLE
2021-1-13

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#424]
151t-distributed stochastic neighbor embedding (t-SNE) is a machine learning algorithm for visualization based on Stochastic Neighbor Embedding originally developed by Sam Roweis and Geoffrey Hinton,[1] where Laurens van der Maaten proposed the t-distributed variant.[2] It is a nonlinear dimensionality reduction technique well-suited for embedding high-dimensional data for visualization in a low-dimensional space of two or three dimensions; (2021-1-6)
152Converts a sentence (string) into a list of words (strings). Extracts the GloVe representation of each word     and averages its value into a single vector encoding the meaning of the sentence.
153sentence -- string, one training example from X
154word_to_vec_map -- dictionary mapping every word in a vocabulary into its 50-dimensional vector representation
155avg -- average vector encoding information about the sentence, numpy-array of shape (50,)
156you can choose a word that exists in the word_to_vec_map and access its .shape field; Hint: you can use any one of the word vectors that you retrieved from the input sentence to find the shape of a word vector; (2021-1-6)
157Pass the average through forward propagation; Compute the cost; Backpropagate to update the softmax parameters; (2021-1-6)
158Step 1: Initialize the embedding matrix as a numpy array of zeros; The embedding matrix has a row for each unique word in the vocabulary; (2021-1-6)
159Step 2: Fill in each row of the embedding matrix with the vector representation of a word; (2021-1-6)
